sentry-core/src/main/java/io/sentry/core/exception/SentryExceptionReader.java

```diff
@@ -0,0 +1,111 @@
+package io.sentry.core.exception;
+
+import io.sentry.core.protocol.Mechanism;
+import io.sentry.core.protocol.SentryException;
+import io.sentry.core.protocol.SentryStackTrace;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public final class SentryExceptionReader {
+
+  /**
+   * Creates a new instance from the given {@code throwable}.
+   *
+   * @param throwable the {@link Throwable} to build this instance from
+   */
+  public static List<SentryException> sentryExceptionReader(final Throwable throwable) {
+    return sentryExceptionReader(extractExceptionQueue(throwable));
+  }
+
+  /**
+   * Creates a new instance from the given {@code exceptions}.
+   *
+   * @param exceptions a {@link Deque} of {@link SentryException} to build this instance from
+   */
+  private static List<SentryException> sentryExceptionReader(final Deque<SentryException> exceptions) {
+    return new ArrayList<>(exceptions);
+  }
+
+  /**
+   * Creates a Sentry exception based on a Java Throwable.
+   * <p>
+   * The {@code childExceptionStackTrace} parameter is used to define the common frames with the child exception
+   * (Exception caused by {@code throwable}).
+   *
+   * @param throwable                Java exception to send to Sentry.
+   * @param childExceptionStackTrace StackTrace of the exception caused by {@code throwable}.
+   * @param exceptionMechanism The optional {@link Mechanism} of the {@code throwable}.
+   *                           Or null if none exist.
+   */
+  private static SentryException sentryExceptionReader(
+    Throwable throwable,
+    StackTraceElement[] childExceptionStackTrace, // TODO: do we need that?
+    Mechanism exceptionMechanism) {
+
+    Package exceptionPackage = throwable.getClass().getPackage();
+    String fullClassName = throwable.getClass().getName();
+
+    SentryException exception = new SentryException();
+
+//    this.exceptionMessage = throwable.getMessage();
+    String exceptionClassName = exceptionPackage != null
+      ? fullClassName.replace(exceptionPackage.getName() + ".", "")
+      : fullClassName;
+
+    String exceptionPackageName = exceptionPackage != null
+      ? exceptionPackage.getName()
+      : null;
+    // TODO: whats about those missing fields? message, classname, packagename, ...
```

Comment:
@bruno-garcia old sdk is setting these fields, but we don't have them

Reply:
message is `value`. `package` is `module`.
It just that the protocol 

Here's the mapping in the old SDK:
https://github.com/getsentry/sentry-java/blob/master/sentry/src/main/java/io/sentry/marshaller/json/ExceptionInterfaceBinding.java#L58